src/common/config_parser.cpp

```diff
@@ -184,6 +184,8 @@ void ConfigParser::addOptionsModel(cli::CLIWrapper& cli) {
     }
   }
 
+  cli.add<bool>("--model-mmap",
+      "Use memory-mapping when loading model (CPU only)");
```

Comment:
Should this option be added only in CPU compilations? I.e. `#if COMPILE_CPU`? I'm not sure what the convention actually is.

Reply:
Also, is this option relevant to the training as well, or can it be added to the translation options only?

Reply:
The mmap code is currently only in translator; the option will only appear when compiled with CPU support and when performing translation.

I added a validation to ensure it is only enabled when cpu-threads is not 0 (i.e. is a non-GPU backend)